File: src/main/java/com/github/kindrat/cassandra/client/filter/fields/NePredicate.java
package com.github.kindrat.cassandra.client.filter.fields;

import com.datastax.driver.core.TypeCodec;
import com.github.kindrat.cassandra.client.ui.DataObject;

public class NePredicate extends RowPredicate {
    public NePredicate(String field, String value) {
        super(field, value);
    }

    @Override
    public boolean test(DataObject data) {
        TypeCodec<?> typeCodec = getColumnCodec(data.get(getField()));

        Object expected = typeCodec.parse(getValue());
        Object actual = data.get(getField());
        if (expected instanceof Comparable && actual instanceof Comparable) {
            //noinspection unchecked
            return Comparable.class.cast(actual).compareTo(expected) != 0;
        } else {
            return expected != actual;
        }
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/fields/NePredicate.kt
package com.github.kindrat.cassandra.client.filter.fields

import com.github.kindrat.cassandra.client.ui.DataObject

class NePredicate(field: String, value: String) : RowPredicate(field, value) {

    override fun test(data: DataObject): Boolean {
        val typeCodec = getColumnCodec(data[field])
        val expected = typeCodec.parse(value)
        val actual = data[field]
        if (expected is Comparable<*> && actual is Comparable<*>) {
            return compareValues(actual, expected) != 0
        } else {
            return expected !== actual
        }
    }
}